src/storage/mvcc/reader/scanner/util.rs

```diff
@@ -81,7 +83,29 @@ where
     assert!(write.short_value.is_none());
     let seek_key = user_key.clone().append_ts(write.start_ts);
     default_cursor.near_seek(&seek_key, &mut statistics.data)?;
-    assert!(default_cursor.valid());
+    if !default_cursor.valid() {
```

Comment:
Shall we always returns Error? If there are data corruptions, the SQL won't have a chance to success, so that we can actually discover it.

Reply:
It is enabled in our test environment, panic would help to debug.

Reply:
I'm curious that how TiDB handle `DefaultNotFound` error?

Reply:
`DefaultNotFound` will convert into `OtherError`, TiDB abort the query when encountering `OtherError`